### DISCUSSION TITLE
With the "Better Villages" mod, bounty board buildings are almost everywhere!

### DISCUSSION THREAD

**Discussion by tp1415926535** [ejektaflex/Bountiful#280] [Mod Compat]
I use [Better Villages - Fabric](https://www.curseforge.com/minecraft/mc-mods/better-village-fabric) mod to improve village buildings.
But now that I want to add this mod, even though I minimized the settings "villageGenFrequency": 1, there are still bounty boards everywhere like this, it is really bad.:
![2024-05-03_13 07 19](https://github.com/ejektaflex/Bountiful/assets/58326584/57b01211-32f8-41be-b051-c3e64ce62917)
The mod should support configurations to be generated by a specified number, not probability.
For what it's worth, I also use the [Waystones (Fabric Edition)](https://www.curseforge.com/minecraft/mc-mods/waystones-fabric) mod, which can ensures that there is only one building of that module in the entire village, so maybe you can refer to that.

**Comment by ejektaflex** [comment]
Hi!

This is because Better Villages uses completely different village jigsaw weights from normal Minecraft. This means that any other mod that injects into the jigsaw system will have the same problem. The best long term solution is for Better Villages to redo their jigsaw pieces with better weights in mind that more closely and accurately reflect the vanilla weights. The best short term solution is for you to use a data pack for Bountiful and Better Villages compatibility - I know compat data packs exist on Modrinth.

As for the Waystones comparison - until 1.20.x, doing what you ask was not an easy feat with mixins. I will look into this possibility for the future though several attempts have proved impractical - it does not mesh well with how Minecraft generates villages.

**Reply by tp1415926535** [reply]
Well, relying on compatibility is always more unstable.
I didn't find this compatibility pack in Modrinth but on curseforge. But I don't know what caused its author to stop updating it a few months ago, stopping at 1.19 while my game is 1.20.4. From what I can see from github's issues, he released a newer version that time  but there's no file for it at the moment. Although it looks like it's fixed after loading latest file, there may be some hidden problems with the different versions.

**Reply by ejektaflex** [reply]
Well, I don't have enough time to maintain compatibility with every village mod - and frankly, if Better Villages were using sensible jigsaw pool weights that mimicked vanilla Minecraft, this wouldn't be a problem in the first place. The onus is more on them as a village overhaul to create a more streamlined experience with mods that add to villages than it is on me to support every custom village solution that has been created.

But, as I mentioned, community support is there! I'm glad to see that the compat pack is still being updated.

**Reply by ejektaflex** [reply]
Additionally, it should be noted that the Better Villages <-> Bountiful compat pack is created by the owner of Better Villages. It looks to me like they're already managing this on their end.
